File: 100dynamic/62.swift
class Solution {
    func uniquePaths(_ m: Int, _ n: Int) -> Int {
        var dp = Array(repeating:0, count: n)
        dp[0] = 1

        for i in 0..<m {
            for j in 1..<n {
                dp[j] = dp[j] + dp[j-1]
            }
        }
        return dp[n-1]
    }
}

File: 100dynamic/62.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [0] * (n)
        dp[0] = 1
        for i in range(0, m):
            for j in range(1, n):
                dp[j] = dp[j-1] + dp[j]
        return dp[-1]